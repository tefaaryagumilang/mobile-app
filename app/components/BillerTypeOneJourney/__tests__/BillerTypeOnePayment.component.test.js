import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {reduxForm} from 'redux-form';
import BillerTypeOnePayment from '../BillerTypeOnePayment.component';

const spy = jest.fn();
const store = createStore(() => ({}));
const DecoratedForm = reduxForm({
  form: 'testForm',
  onSubmit: spy
})(BillerTypeOnePayment);

describe('BillerTypeOnePayment Component: PaymentSelection', () => {
  it('renders correctly', () => {
    const formValues = {};
    const accounts = [];
    const tree = renderer.create(
      <Provider store={store}>
        <DecoratedForm formValues={formValues} accounts={accounts} />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
