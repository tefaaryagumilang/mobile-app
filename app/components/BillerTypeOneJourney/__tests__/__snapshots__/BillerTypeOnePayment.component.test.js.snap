// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillerTypeOnePayment Component: PaymentSelection renders correctly 1`] = `
<KeyboardAwareScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  enableOnAndroid={true}
  extraScrollHeight={100}
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View
      style={
        Array [
          Object {
            "padding": 20,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Array [
                  Object {
                    "color": "#4ed07d",
                    "marginTop": 5,
                  },
                  Object {
                    "marginRight": 10,
                  },
                ],
                Object {
                  "fontFamily": "simas",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "roboto",
                "fontSize": 22,
              }
            }
          >
            Jumlah total
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 25,
          }
        }
      />
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#949494",
              "borderRadius": 5,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "fontWeight": "700",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 20,
                    "marginLeft": 10,
                    "marginRight": -15,
                  },
                ]
              }
            >
              Rp
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={false}
            format={[Function]}
            label="Jumlah uang"
            normalize={[Function]}
            placeholder="0"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "fontWeight": "700",
                },
                Object {
                  "color": "#000000",
                  "flex": 1,
                  "fontFamily": "roboto",
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="--"
          />
        </View>
      </View>
      <View />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ECECEC",
          "height": 1,
          "marginTop": 10,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "padding": 20,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "color": "#c38a05",
                    },
                    Object {
                      "marginRight": 10,
                    },
                  ],
                  Object {
                    "fontFamily": "simas",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "roboto",
                  "fontSize": 22,
                }
              }
            >
              Pilih rekening sumber
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 25,
            }
          }
        />
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "roboto",
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              Pilih rekening sumber
            </Text>
          </View>
          <View>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontFamily": "simas",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ECECEC",
          "height": 1,
          "marginTop": 10,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "padding": 20,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "padding": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FA3F70",
              "borderRadius": 30,
              "flexDirection": "row",
              "height": 47,
              "justifyContent": "center",
              "paddingHorizontal": 15,
            },
            undefined,
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "Roboto",
              },
            ]
          }
        >
          BERIKUTNYA
        </Text>
      </View>
    </View>
  </View>
</KeyboardAwareScrollView>
`;
